src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people.filter(man => {
+    return century
+      ? man.sex === 'm' && Math.ceil(man.died / 100) === century
+      : man.sex === 'm';
+  });
+
+  const sumOfAges = filteredMen
+    .map(man => man.died - man.born)
+    .reduce((sum, currentAge) => sum + currentAge);
+
+  return sumOfAges / filteredMen.length;
```

Comment:
Would be nice to move it to a separate function, considering that the same operation is done in `calculateWomenAverageAge`

Reply:
fix @glugovsky comment

Reply:
added, thx)